File: C#/Ejercicio6.cs
using System;

void InsertionSort(int[] arr)
{
    int n = arr.Length;
    for (int i = 1; i < n; i++)
    {
        int actual = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > actual)
        {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = actual;
    }
}

int[] arr = {64, 34, 25, 12, 22, 11, 90};
Console.WriteLine("=== INSERTION SORT EN C# SCRIPT ===");
Console.WriteLine("Array inicial: [" + string.Join(", ", arr) + "]");

InsertionSort(arr);

Console.WriteLine("Array final: [" + string.Join(", ", arr) + "]");

File: JavaScript/Ejercicio6.js
function printArray(arr) {
    return `[${arr.join(', ')}]`;
}

function inertionSort(arr) {
    const n = arr.length;
    for (let i = 1; i < n; i++) {
        let actual = arr[i];
        let j = i - 1;
        while (j >= 0 && arr[j] > actual) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = actual;
    }
}

function main() {
    const arr = [54, 24, 15, 2, 12, 1, 80];

    console.log("=== INSERTION SORT EN JAVASCRIPT ===");
    console.log(`Array inicial: ${printArray(arr)}`);

    inertionSort(arr);

    console.log(`Array final: ${printArray(arr)}`);
}

main();